src/python/pants/init/options_initializer.py

```diff
@@ -111,42 +103,22 @@ def _install_options(self, options_bootstrapper, build_configuration):
     for optionable_cls in distinct_optionable_classes:
       optionable_cls.register_options_on_scope(options)
 
-    # Make the options values available to all subsystems.
-    Subsystem.set_options(options)
-
     return options
 
-  def setup(self, init_logging=True):
-    """Initializes logging, loads backends/plugins and parses options.
-
-    :param bool init_logging: Whether or not to initialize logging as part of setup.
-    :returns: A tuple of (options, build_configuration).
-    """
-    global_bootstrap_options = self._options_bootstrapper.get_bootstrap_options().for_global_scope()
+  @classmethod
+  def create(cls, options_bootstrapper, build_configuration, init_subsystems=True):
+    global_bootstrap_options = options_bootstrapper.get_bootstrap_options().for_global_scope()
 
     if global_bootstrap_options.pants_version != pants_version():
       raise BuildConfigurationError(
         'Version mismatch: Requested version was {}, our version is {}.'
         .format(global_bootstrap_options.pants_version, pants_version())
       )
 
-    # Get logging setup prior to loading backends so that they can log as needed.
-    if init_logging:
-      self._setup_logging(global_bootstrap_options.quiet,
-                          global_bootstrap_options.level,
-                          global_bootstrap_options.logdir)
-
-    # Conditionally load backends/plugins and materialize a `BuildConfiguration` object.
-    if not self._has_build_configuration():
-      build_configuration = self._load_plugins(self._working_set,
-                                               global_bootstrap_options.pythonpath,
-                                               global_bootstrap_options.plugins,
-                                               global_bootstrap_options.backend_packages)
-      self._set_build_configuration(build_configuration)
-    else:
-      build_configuration = self._get_build_configuration()
-
     # Parse and register options.
-    options = self._install_options(self._options_bootstrapper, build_configuration)
+    options = cls._install_options(options_bootstrapper, build_configuration)
```

Comment:
It looks like maybe this method is side-effect free? If so, renaming it from `install` to `construct` or something would be good.

Reply:
fixed